locator.go

```diff
@@ -0,0 +1,468 @@
+	if err != nil {
+		return nil, err
+	}
+	texts := innerTexts.([]interface{})
+	result := make([]string, len(texts))
+	for i := range texts {
+		result[i] = texts[i].(string)
+	}
+	return result, nil
+}
+
+func (l *locatorImpl) AllTextContents() ([]string, error) {
+	textContents, err := l.frame.EvalOnSelectorAll(l.selector, "ee => ee.map(e => e.textContent || '')")
+	if err != nil {
+		return nil, err
+	}
+	texts := textContents.([]interface{})
+	result := make([]string, len(texts))
+	for i := range texts {
+		result[i] = texts[i].(string)
+	}
+	return result, nil
+}
+
+func (l *locatorImpl) BoundingBox(options ...LocatorBoundingBoxOptions) (*LocatorBoundingBoxResult, error) {
+	var option PageWaitForSelectorOptions
+	if len(options) == 1 {
+		option.Timeout = options[0].Timeout
+	}
+
+	var result *LocatorBoundingBoxResult
+	err := l.withElement(func(handle ElementHandle) error {
+		rect, err := handle.BoundingBox()
+		if err != nil {
+			return err
+		}
+		result = &LocatorBoundingBoxResult{
+			X:      float64(rect.X),
+			Y:      float64(rect.Y),
+			Width:  float64(rect.Width),
+			Height: float64(rect.Height),
+		}
+		return nil
+	}, option)
+
+	if err != nil {
+		return nil, err
+	}
+
+	return result, nil
+}
+
+func (l *locatorImpl) Check(options ...FrameCheckOptions) error {
+	return l.frame.Check(l.selector, options...)
+}
+
+func (l *locatorImpl) Click(options ...PageClickOptions) error {
+	return l.frame.Click(l.selector, options...)
+}
+
+func (l *locatorImpl) Count() (int, error) {
+	return l.frame.queryCount(l.selector)
+}
+
+func (l *locatorImpl) Dblclick(options ...FrameDblclickOptions) error {
+	return l.frame.Dblclick(l.selector, options...)
+}
+
+func (l *locatorImpl) DispatchEvent(typ string, eventInit interface{}, options ...PageDispatchEventOptions) error {
+	return l.frame.DispatchEvent(l.selector, typ, eventInit, options...)
+}
+
+func (l *locatorImpl) DragTo(target Locator, options ...FrameDragAndDropOptions) error {
+	if len(options) == 1 {
+		strict := true
+		options[0].Strict = &strict
+	}
+	return l.frame.DragAndDrop(l.selector, target.(*locatorImpl).selector, options...)
+}
+
+func (l *locatorImpl) ElementHandle(options ...LocatorElementHandleOptions) (ElementHandle, error) {
+	strict := true
+	option := PageWaitForSelectorOptions{
+		State:  WaitForSelectorStateAttached,
+		Strict: &strict,
+	}
+	if len(options) == 1 {
+		option.Timeout = options[0].Timeout
+	}
+	return l.frame.WaitForSelector(l.selector, option)
+}
+
+func (l *locatorImpl) ElementHandles() ([]ElementHandle, error) {
+	return l.frame.QuerySelectorAll(l.selector)
+}
+
+func (l *locatorImpl) Evaluate(expression string, arg interface{}, options ...LocatorEvaluateOptions) (interface{}, error) {
+	var option PageWaitForSelectorOptions
+	if len(options) == 1 {
+		option.Timeout = options[0].Timeout
+	}
+
+	var result interface{}
+	err := l.withElement(func(handle ElementHandle) (err error) {
```

Comment:
in theory we could change withElement that it returns `(error, interface{})` so we don't need to have the outer variable assignment. Not sure tho if thats better!